xee/ext.py

```diff
@@ -420,13 +420,23 @@ def project(self, bbox: types.BBox) -> types.Grid:
         appropriate region of data to return according to the Array's configured
         projection and scale.
     """
-    # The origin of the image is in the top left corner. X is the minimum value
-    # and Y is the maximum value.
-    x_origin, _, _, y_origin = self.bounds  # x_min, x_max, y_min, y_max
+    # The origin of the image is in the top left corner.
+    x_min, y_min, x_max, y_max = self.bounds
     x_start, y_start, x_end, y_end = bbox
     width = x_end - x_start
     height = y_end - y_start
 
+    # Found the actual coordinates of the first or last point of the bbox based on the pos & neg scale in the actual image.
+    translateX = (
```

Comment:
We should factor out the common parts of this expression (e.g. `translateX = self.scale_x * x_start + (x_min if self.scale_x > 0 else x_max)`).

Similar for `translateY`.